core/src/main/scala/org/broadinstitute/dsde/rawls/util/WorkspaceSupport.scala

```diff
@@ -39,6 +32,13 @@ trait WorkspaceSupport {
   import dataSource.dataAccess.driver.api._
 
   // Access/permission helpers
+  def userEnabledCheck(userInfo: UserInfo): Future[Unit] =
+    samDAO.getUserStatus(userInfo) flatMap {
+      case Some(user) =>
+        if (user.enabled) Future.successful()
+        else Future.failed(new UserDisabledException(StatusCodes.Unauthorized, "Unauthorized"))
+      case None => Future.failed(new UserDisabledException(StatusCodes.Unauthorized, "Unauthorized"))
```

Comment:
optional, I wouldn't block the PR on this … if you felt like consolidating:
```scala
      case Some(user) if (user.enabled) => Future.successful()
      case _ => Future.failed(new UserDisabledException(StatusCodes.Unauthorized, "Unauthorized"))
```

Reply:
Good call, implemented.